merlin/dtype/__init__.py

```diff
@@ -0,0 +1,122 @@
+#
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import sys
+from copy import copy
+from dataclasses import dataclass
+from enum import Enum
+from types import ModuleType
+from typing import Any, Dict, Optional, Tuple
+
+import numpy as np
+
+
+class ElementType(Enum):
+    Bool = "bool"
+    Int = "int"
+    UInt = "uint"
+    Float = "float"
+    String = "string"
+    DateTime = "datetime"
+    Object = "object"
+
+
+@dataclass
+class DType:
+    name: str
+    # TODO: Rename elemsize to bits or bytes for clarity
+    elemtype: ElementType
+    elemsize: Optional[int] = None
+    signed: Optional[bool] = None
+    shape: Optional[Tuple] = None
+
+    @property
+    def is_list(self):
+        return self.shape is not None and len(self.shape) > 1
+
+    @property
+    def is_ragged(self):
+        return self.is_list and None in self.shape
+
+
+int32 = DType("int32", ElementType.Int, 32, signed=True)
+int64 = DType("int64", ElementType.Int, 64, signed=True)
+uint32 = DType("uint32", ElementType.UInt, 32)
+uint64 = DType("uint64", ElementType.UInt, 64)
+float32 = DType("float32", ElementType.Float, 32, signed=True)
```

Comment:
Would we have to include integers and floats at lower bits like 16?

Reply:
Yeah, that's actually required in order to make the NVT tests pass. The first commit on this branch was a rough sketch, but I'm fleshing it out as I test it with the other Merlin repos.